core/src/main/java/com/sequenceiq/cloudbreak/service/upgrade/ClusterUpgradeAvailabilityService.java

```diff
@@ -157,12 +157,17 @@ private UpgradeV4Response checkForUpgrades(Stack stack, boolean lockComponents)
         try {
             LOGGER.info(String.format("Retrieving images for upgrading stack %s", stack.getName()));
             com.sequenceiq.cloudbreak.cloud.model.Image currentImage = getImage(stack);
-            CloudbreakImageCatalogV3 imageCatalog = getImagesFromCatalog(stack.getWorkspace(), currentImage);
-            Image image = getCurrentImageFromCatalog(currentImage.getImageId(), imageCatalog);
-            ImageFilterParams imageFilterParams = imageFilterParamsFactory.create(image, lockComponents, stack);
-            ImageFilterResult filteredImages = filterImages(stack.getCreator().getTenant().getName(), imageCatalog, stack.cloudPlatform(), imageFilterParams);
-            LOGGER.info(String.format("%d possible image found for stack upgrade.", filteredImages.getAvailableImages().getCdhImages().size()));
-            upgradeOptions = createResponse(image, filteredImages, stack.getCloudPlatform(), stack.getRegion(), currentImage.getImageCatalogName());
+            if (currentImage.getImageCatalogUrl() == null) {
+                upgradeOptions.setReason(String.format("Upgarde is not supported by clusters with custom image"));
```

Comment:
I think a TODO or a FIXME would be nice here - currently it looks like we don't even plan to support it. Or at least add a "yet" to the end of the message :)

Reply:
It was there but I have removed it as the message states the current behavior :-) But yes, I can agree.